### PR TITLE
Implement tsconfig-files method

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Relates to #2362

### CODE REVIEW

File: src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeClientTests.cs [hunk 1]
```diff
@@ -163,6 +163,52 @@ public void NewTsConfig_IncorrectResponse_Throws()
             act.Should().ThrowExactly<InvalidOperationException>();
         }
 
+        [TestMethod]
+        public async Task TSConfigFiles_HttpWrapperCalledWithCorrectArguments()
```
Comment (duncanp-sonar) [inline src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeClientTests.cs:167]: Same set of tests as for `AnalyseJs` (cut-and-paste then modified)

File: src/TypeScript/EslintBridgeClient/Contract/AnalysisResponse.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient.Contract
     internal class AnalysisResponse
     {
         // Other fields omitted
-        // i.e. Hightlight[], HighlightedSymbol[], Metrics, CpdToken[]
+        // i.e. Highlight[], HighlightedSymbol[], Metrics, CpdToken[]
```
Comment (duncanp-sonar) [inline src/TypeScript/EslintBridgeClient/Contract/AnalysisResponse.cs:29]: Miscellaneous - fixing typo

File: src/TypeScript/EslintBridgeClient/EslintBridgeClient.cs [hunk 3]
```diff
@@ -115,6 +121,18 @@ public async Task NewTsConfig(CancellationToken cancellationToken)
             }
         }
 
+        public async Task<TSConfigResponse> TsConfigFiles(string tsConfigFilePath, CancellationToken cancellationToken)
+        {
+            var responseString = await httpWrapper.PostAsync(BuildServerUri("tsconfig-files"), tsConfigFilePath, cancellationToken);
+
+            if (string.IsNullOrEmpty(responseString))
```
Comment (duncanp-sonar) [inline src/TypeScript/EslintBridgeClient/EslintBridgeClient.cs:128]: Same error handling as for `AnalyzeJS`. This wrapper will throw if we get an invalid response. The response could contain error codes, but it's up to the caller to handle them.

File: src/TypeScript/EslintBridgeClient/Contract/TSConfigResponse.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    internal class TSConfigResponse
+    {
+        /// <summary>
+        /// File paths to source files referenced by the tsconfig file
+        /// </summary>
+        [JsonProperty("files")]
+        public IEnumerable<string> Files { get; set; }
+
+        /// <summary>
+        /// File paths of referenced projects that should be resolved by calling "tsconfig-files"
```
Comment (rita-sonarsource) [inline src/TypeScript/EslintBridgeClient/Contract/TSConfigResponse.cs:35]: I'm not sure I understand this comment. Will we ever need to use this property? Or do we only care about "Files"?

File: src/TypeScript/EslintBridgeClient/Contract/TSConfigResponse.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient.Contract
+{
+    internal class TSConfigResponse
+    {
+        /// <summary>
+        /// File paths to source files referenced by the tsconfig file
+        /// </summary>
+        [JsonProperty("files")]
+        public IEnumerable<string> Files { get; set; }
```
Comment (rita-sonarsource) [inline src/TypeScript/EslintBridgeClient/Contract/TSConfigResponse.cs:32]: Maybe "ReferencedFiles"? Not sure if it's better.

**Review by rita-sonarsource — APPROVED** [review]

File: src/TypeScript/EslintBridgeClient/Contract/TSConfigResponse.cs (hunk 5 above)
Comment (duncanp-sonar) [reply src/TypeScript/EslintBridgeClient/Contract/TSConfigResponse.cs:32]: I'd rather stick to the same names as in the TS and Java code unless there's a strong reason to change them.

File: src/TypeScript/EslintBridgeClient/Contract/TSConfigResponse.cs (hunk 4 above)
Comment (duncanp-sonar) [reply src/TypeScript/EslintBridgeClient/Contract/TSConfigResponse.cs:35]: I was going to exclude the property, but I had a look at the Java code and it is used. The Java code parses the response and calls `tsconfig-files` for each value in `ProjectReferences` (which suggests that a "project reference" is another tsconfig file).

**Pull request merged**